pallets/pallet-governance/src/lib.rs

```diff
@@ -0,0 +1,407 @@
+				let pass: Percent = Percent::from_rational(count_yays as u32, count_council as u32);
+				let fail: Percent = Percent::from_rational(count_nays as u32, count_council as u32);
+
+				if pass >= T::MajorityThreshold::get() {
+					ProposalsToResolve::<T>::get().insert((proposal, true));
+				}
+
+				if fail >= Percent::one().saturating_sub(T::MajorityThreshold::get()) {
+					ProposalsToResolve::<T>::get().insert((proposal, false));
+				}
+			} else {
+				// return error, can not reach here.
+			}
+
+			// Emit Event: VoteCasted
+			Self::deposit_event(Event::<T>::VoteCasted { who: id, proposal, approve });
+
+			Ok(())
+		}
+
+		// Requires Root origin - sets the current authorities
+		#[pallet::call_index(2)]
+		#[pallet::weight(T::WeightInfo::force_rotate_authorities())]
+		pub fn force_rotate_authorities(
+			origin: OriginFor<T>,
+			new_members: Vec<T::AccountId>,
+		) -> DispatchResult {
+			// ensure root
+			ensure_root(origin)?;
+			Self::rotate_authorities(new_members);
+			Ok(())
+		}
+
+		// Requires Governance origin - sets the current authorities
+		#[pallet::call_index(3)]
+		#[pallet::weight(T::WeightInfo::council_rotate_authorities())]
+		pub fn council_rotate_authorities(
+			origin: OriginFor<T>,
+			new_members: Vec<T::AccountId>,
+		) -> DispatchResult {
+			// ensure governance
+			T::EnsureGovernance::ensure_origin(origin)?;
+
+			Self::rotate_authorities(new_members);
+			Ok(())
+		}
+	}
+
+	/// Custom governance origin
+	pub struct EnsureGovernance;
+
+	/// Implementation for EnsureOrigin trait for custom EnsureGovernance struct.
+	/// We use this to execute extrinsic by a governance origin.
+	impl<OuterOrigin> EnsureOrigin<OuterOrigin> for EnsureGovernance
+	where
+		OuterOrigin: Into<Result<RawOrigin, OuterOrigin>> + From<RawOrigin>,
+	{
+		type Success = ();
+
+		fn try_origin(o: OuterOrigin) -> Result<Self::Success, OuterOrigin> {
+			match o.into() {
+				Ok(o) => match o {
+					RawOrigin::GovernanceApproval => Ok(()),
+				},
+				Err(o) => Err(o),
+			}
+		}
+
+		#[cfg(feature = "runtime-benchmarks")]
+		fn try_successful_origin() -> Result<OuterOrigin, ()> {
+			Ok(RawOrigin::GovernanceApproval.into())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		/// Dispatches a call from the governance origin, with transactional semantics, ie. if the
+		/// call dispatch returns `Err`, rolls back any storage updates.
+		fn dispatch_governance_call(
+			call: <T as Config>::RuntimeCall,
+		) -> DispatchResultWithPostInfo {
+			with_transaction(move || {
+				match call.dispatch_bypass_filter(RawOrigin::GovernanceApproval.into()) {
+					r @ Ok(_) => TransactionOutcome::Commit(r),
+					r @ Err(_) => TransactionOutcome::Rollback(r),
+				}
+			})
+		}
+
+		/// Increased proposal ID.
+		fn next_proposal_id() -> ProposalId {
+			NextProposalId::<T>::mutate(|id| {
+				*id = id.saturating_add(1);
+				*id
+			})
+		}
+
+		fn rotate_authorities(authorities: Vec<T::AccountId>) {
```

Comment:
should rename to do_rotate_authorities.
Usually the internal function that do something without the checks are named do_something